spec/type_references_and_ownership.md

```diff
@@ -0,0 +1,675 @@
+deploy a single ARM template with the entire resource graph. Each resource will be done in its own deployment (with
+a `dependsOn` specified if required).
+
+### Aren't there going to be races in resource creation?
+Yes. If you have a complex hierarchy of resources (where resources have involved relationships between one another) and submit 
+all of their YAMLs to the operator at the same time it is likely that some requests when sent to ARM will fail because of missing dependencies.
+Those resources that failed to deploy initially will be in an unprovisioned state in Kubernetes, and eventually
+all the resources will be created through multiple iterations of the reconciliation loop. 
+
+### Aren't there going to be races in resource deletion?
+Yes. `Owner` as discussed in this specification is informing Kubernetes _how Azure behaves_. The fact that 
+a `ResourceGroup` is the owner of a `VirtualMachineScaleSet` means that when the `ResourceGroup` is deleted in 
+Azure, the `VirtualMachineScaleSet` will be too. 
+
+This means that practically speaking, we don't need Kubernetes garbage collection to 
+perform deletion of resources in Azure. Azure is already going to do that automatically. We need Kubernetes garbage collection
+to easily maintain sync with Azure.
+
+As far as implementation goes this just means that when we are performing deletes in the generic controller and the 
+resource is already deleted in Azure we just swallow that error and allow the Kubernetes object to be deleted.
+
+### What exactly happens when a resource with an `Owner` is created?
+Once the resource has been accepted by the various admissions controllers and has been cofirmed to match 
+the structural schema defined in the CRD, the generic controller will attempt to look up
+the owning resource in etcd (or in ARM if it's an `AzureReference`).
+
+If the generic controller finds the owning resource, it updates the `ownerReference` in the object metadata
+to include the `uid` of the owning resource and then submits an ARM template to ARM using the 
+name of the owner and the name of the resource to build the name specified in the ARM template. It will
+include the name of the owner in the `dependsOn` field. 
+
+### What happens if an owning resource is deleted and immediately recreated?
+Kubernetes garbage collection is based on object `uid`'s. As discussed above we bind to that `uid` on 
+dependent resource creation. If a resource is deleted and then recreated Kubernetes will still understand
+that the new resource is fundamentally different than the old resource and garbage collection will happen
+as expected. The result will be that there is a new owning resource but all of its dependent resources were 
+deleted (in Azure and in k8s).
+
+## TODOs
+- How can we allow customers to easily find all dependents for a particular owner (i.e. all subnets of a vnet) using `kubectl`?
+- Cross subscription refs?
```

Comment:
Note here that these aren't supported by Azure.

Reply:
They are in some cases I believe? https://stackoverflow.com/questions/61001675/azure-vnet-cross-subscription-peering

Reply:
Right, but in general we can’t do it. That example is specific to VNets and needed for their specific functionality.

Reply:
Ahh, got it. Will fix up document to clarify.